public/video-ui/src/components/Video/Display.js

```diff
@@ -8,69 +8,89 @@ import YoutubeMetaData from '../YoutubeMetaData/YoutubeMetaData';
 import VideoPoster from '../VideoPoster/VideoPoster';
 import GridImageSelect from '../utils/GridImageSelect';
 import Icon from '../Icon';
+import {validate} from '../../constants/videoEditValidation';
 
 class VideoDisplay extends React.Component {
 
-  state = {
-    metadataEditable: false,
-    youtubeEditable: false
-  };
-
   componentWillMount() {
     this.props.videoActions.getVideo(this.props.params.id);
   }
 
   saveVideo = () => {
     this.props.videoActions.saveVideo(this.props.video);
+  }
 
-    this.setState({
-      editable: false
-    });
+  saveAndUpdateVideo = (video) => {
+    this.props.videoActions.saveVideo(video);
   };
 
-  saveAndUpdateVideo = (video) => {
-    this.props.videoActions.debouncedSaveVideo(video);
+  updateVideo = (video) => {
+    this.props.videoActions.updateVideo(video);
   };
 
   selectVideo = () => {
     window.parent.postMessage({atomId: this.props.video.id}, '*');
   };
 
-  manageEditingState = (value, property) => {
+  manageEditingState = (property) => {
 
-    if (property === 'metadata') {
+    if (this.props.editState[property]) {
 
-      this.setState({
-        metadataEditable: value
-      });
-
-    } else if (property === 'youtube') {
-
-      this.setState({
-        youtubeEditable: value
-      });
+      this.saveAndUpdateVideo(this.props.video);
     }
-  };
 
-  disableEditing = () => {
-    this.setState({
-      editable: false
-    });
+    const newEditableState = Object.assign(this.props.editState, {[property]: !this.props.editState[property]});
+    this.props.videoActions.updateVideoEditState(newEditableState);
   };
 
   cannotEditStatus = () => {
     return this.props.video.expiryDate <= Date.now();
   };
 
-  renderEditButton = (editable, property) => {
+  cannotOpenEditForm = () => {
+    if (this.props.editState && (this.props.editState.metadataEditable || this.props.editState.youtubeEditable)) {
+      return true;
+    }
+
+    return false;
+  }
+
+
+  cannotCloseEditForm = (property) => {
+
+    const formFields = [];
+    if (property === 'metadataEditable') {
+      formFields.splice(0, 0, 'title', 'category');
```

Comment:
`.concat` was bad advice as it returns a new array. The spread operator is the right way to go `formFields.push(...['title', 'category'])`.